tests/device_tests/test_device_e2e.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+"""Test file for binary running on devices"""
+import glob
+import os
+import sys
+import unittest
+
+
+class DeviceE2eTest(unittest.TestCase):
+    """Test Case of Device E2E Test."""
+
+    def _get_param(self, test_case, input_path, lib_name):
+        test_case_dir = os.path.join(input_path, test_case)
+        output_dir = glob.glob(os.path.join(test_case_dir, "export/*/*/output"))
+        if output_dir:
+            output_dir = output_dir[0]
+        else:
+            raise Exception("No such directory")
+        test_data_dir = os.path.join(os.path.dirname(output_dir), "inference_test_data")
+        model_dir = os.path.join(output_dir, "models")
+        lib_dir = os.path.join(model_dir, "lib")
+        return {
+            "python_path": os.path.join(output_dir, "python"),
+            'image': os.path.join(test_data_dir, "raw_image.png"),
+            'model': os.path.join(lib_dir, lib_name),
+            'config': os.path.join(model_dir, "meta.yaml"),
+            }
+
+    def _get_test_cases(self, input_path, lib_name):
+        return [[case, self._get_param(case, input_path, lib_name)] for case in os.listdir(input_path)]
+
+    def _run(self, python_path, image, model, config):
+        sys.path.append(python_path)
```

Comment:
Maybe we need to initialize `sys.path` for every test case to avoid the conflict of the module name `run` among all test cases.
```py
class DeviceE2eTest(unittest.TestCase):
    sys_path_default = copy.deepcopy(sys.path)

    def _run():
        sys.path = copy.deepcopy(sys_path_default) + [python_path]
```

Reply:
@tfujiwar 
Nice review! Thanks a lot! It would cause a bug, I didn' t noticed.
I fixed it 👍